Create App initial view - homework_1 v.1

Nyskelo opened the pull request:
- Components are shown as function components (.jsx extensions).
- Presented all input and buttons as reusable components.
- Installed eslint, prettier and pre-commit hooks.
- Implemented title / id search unctionality. Reset search result if search bar is empty.
- Added the possibility to add a course title / description / duration (format "hh:mm").
- Added logic to create a new author.

Nyskelo committed:
Create App initial view - homework_1 v.1

- Components are shown as function components (.jsx extensions).
- Presented all input and buttons as reusable components.
- Installed eslint, prettier and pre-commit hooks.
- Implemented title / id search unctionality. Reset search result if
  search bar is empty.
- Added the possibility to add a course title / description / duration
  (format "hh:mm").
- Added logic to create a new author.

EpamFanAnton reviewed (changes requested):
While testing your app I found:
1. I can't create new authors.

Nyskelo committed:
Fix issues homework-1 v.2

1.Change regex to make it easy to create an author.

2.Added the ternarny operator.

3.Fixed by using a setter without a prev.

4.In the useValidation custom hook, the syntax is changed to be similar
  to that of the ReactHooks.

EpamFanAnton reviewed (approved):
20
